work24/register.js

```diff
@@ -0,0 +1,51 @@
+const modal = document.getElementById('js-modal')
+const mask = document.getElementById('js-mask')
+const checkbox = document.getElementById('js-checkbox')
+const agreeText = document.getElementById('js-agreeText')
+const closeIcon = document.getElementById('js-closeIcon')
+const closeBtn = document.getElementById('js-closeBtn')
+const submitBtn = document.getElementById('js-submitBtn')
+
+agreeText.addEventListener('click', () => {
+  modalOpen()
+})
+
+closeIcon.addEventListener('click', () => {
+  modalClose()
+})
+
+closeBtn.addEventListener('click', () => {
+  modalClose()
+})
+
+submitBtn.addEventListener('click', (e) => {
+  e.preventDefault()
+  if (checkbox.checked) {
+    location.href = './register-done.html'
+  } else {
+    alert('利用規約を最後まで読んでから送信ボタンを押してください')
+  }
+})
+
+modal.onscroll = function () {
+  const modalHeight = modal.clientHeight
+  const modalScrollHeight = modal.scrollHeight
+  const modalScrollTop = modal.scrollTop
```

Comment:
ここ`this`使えます(私も先生に教えて頂きました)
```suggestion
  const modalHeight = this.clientHeight
  const modalScrollHeight = this.scrollHeight
  const modalScrollTop = this.scrollTop
```

Reply:
なるほどです。
思ったのですが、thisを使うことのメリットってなんなんでしょうか？

Reply:
先生のレビューは

> `this`でも可

とのことでしたのでどちらでも間違いではないという認識です。

Reply:
> 思ったのですが、thisを使うことのメリットってなんなんでしょうか？

今回の場合読みやすさじゃないですかね

Reply:
かしこまりました！
同じような場合にthis使ってみようと思います！
ありがとうございます。